src/fsharp/vs/ServiceParamInfoLocations.fs

```diff
@@ -77,66 +77,65 @@ module internal NoteworthyParamInfoLocationsImpl =
 
     let getTypeName(synType) =
         match synType with
-        | SynType.LongIdent(LongIdentWithDots(ids,_)) -> ids |> List.map textOfId
+        | SynType.LongIdent(LongIdentWithDots(ids,_)) -> ids |> pathOfLid
         | _ -> [""] // TODO type name for other cases, see also unit test named "ParameterInfo.LocationOfParams.AfterQuicklyTyping.CallConstructorViaLongId.Bug94333"
 
+    let handleSingleArg traverseSynExpr (pos, synExpr, parenRange, rpRangeOpt : _ option) =
+        let inner = traverseSynExpr synExpr
+        match inner with
+        | None ->
+            if AstTraversal.rangeContainsPosEdgesExclusive parenRange pos then
+                Found (parenRange.Start, [(parenRange.End, getNamedParamName synExpr)], rpRangeOpt.IsSome), None
+            else
+                NotFound, None
+        | _ -> NotFound, None
+
     // This method returns a tuple, where the second element is
     //     Some(cache)    if the implementation called 'traverseSynExpr expr', then 'cache' is the result of that call
     //     None           otherwise
     // so that callers can avoid recomputing 'traverseSynExpr expr' if it's already been done.  This is very important for perf, 
     // see bug 345385.
-    let rec astFindNoteworthyParamInfoLocationsSynExprExactParen traverseSynExpr pos expr =
-        let handleSingleArg(synExpr, parenRange, rpRangeOpt : _ option) =
-            let inner = traverseSynExpr synExpr
-            match inner with
-            | None ->
-                if AstTraversal.rangeContainsPosEdgesExclusive parenRange pos then
-                    let r = parenRange.Start, [parenRange.End, getNamedParamName synExpr], rpRangeOpt.IsSome
-                    Trace.PrintLine("LanguageServiceParamInfo", fun () -> sprintf "Found single arg paren range %+A from %+A" r expr)
-                    Some r, None
-                else
-                    None, None
-            | _ -> None, None
-
+    let rec searchSynArgExpr traverseSynExpr pos expr =
```

Comment:
I cleaned up this code a bit to make it more obvious what's going on, allowing me to add the new rule for ``ID<``